encoding/x/protobuf/types.go

```diff
@@ -31,17 +31,20 @@ import (
 	"github.com/gogo/protobuf/proto"
 )
 
-// Encoding is the name of this encoding.
-const Encoding transport.Encoding = "protobuf"
+const (
+	// Encoding is the name of this encoding.
+	Encoding transport.Encoding = "protobuf"
 
-// GetApplicationError returns the application error from the server, if present.
+	rawResponseHeaderKey = "yarpc-protobuf-raw-response"
+)
+
+// SetRawResponse will set rawResponseHeaderKey to "true".
 //
-// TODO: this has overlap with IsApplicationError
-func GetApplicationError(err error) error {
-	if applicationError, ok := err.(*applicationError); ok {
-		return applicationError
-	}
-	return nil
+// rawResponseHeaderKey is a header key attached to either a request or
+// response that signals a UnaryHandler to not encode an application error
+// inside a wirepb.Response object, instead marshalling the actual response.
+func SetRawResponse(headers transport.Headers) transport.Headers {
```

Comment:
See https://github.com/yarpc/yarpc-go/issues/905, and the corresponding comment I added on `encoding/x/protobuf/inbound.go`. There is no other way to accomplish this without a two-way dependency between protobuf and grpc as we stand. In the future, the goal will be that protobuf will only be a raw response, and transports are able to understand application level errors @kriskowal @willhug